File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        water = 0
        l,r = 0, len(height)-1

        while(l<r):
            h = min(height[l],height[r])
            w = r-l
            area=w*h
            water=max(water,area)

            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return water

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length-1;
        int water = 0;

        while(l<r){
            int w = (r-l);
            int h=Math.min(height[l],height[r]);
            int area=w*h;
            water=Math.max(water,area);
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return water;
    }
}